src/components/RelationIcon/RelationIcon.tsx

```diff
@@ -16,7 +16,12 @@ const RelationIcon = ({ applicationName, applications }: Props) => {
   const charmId =
     "charm-url" in application
       ? application["charm-url"]
-      : application["charm"];
+      : "charm" in application
```

Comment:
Previously the application type was marked as always having all attributes, but there was a case where this might not be true, so I had to update everywhere to check if the application was the full object.